### PR TITLE
feat(input-time-picker): add focus trap support

### PR DESCRIPTION
**Pull request by Elijbet**
**Description**
Related Issue: https://github.com/Esri/calcite-components/issues/6833

### <ins>Summary<ins>
Adds `focus-trap` to `time-picker` dialog to improve a11y.

### CODE REVIEW

File: src/components/input-time-picker/input-time-picker.tsx [hunk 1]
```diff
@@ -449,6 +500,11 @@ export class InputTimePicker
     }
   };
 
+  private restoreInputFocus(): void {
```
Comment (jcfranco) [inline src/components/input-time-picker/input-time-picker.tsx:503]: I think you can drop this method and call `calciteInputEl.setFocus()` where needed instead.

File: src/components/input-time-picker/input-time-picker.tsx [hunk 2]
```diff
@@ -84,12 +92,39 @@ export class InputTimePicker
...
+      activateFocusTrap(this, {
+        onActivate: () => {
+          if (this.focusOnOpen) {
+            this.calciteTimePickerEl.setFocus();
+            this.focusOnOpen = false;
+          }
+        }
+      });
+    } else {
+      deactivateFocusTrap(this);
```
Comment (jcfranco) [inline src/components/input-time-picker/input-time-picker.tsx:105]: Similar to `activateFocusTrap` above, can you pass an options object with `onDeactivate` where you set the following lines (restore focus and set `focusOnOpen`)? Just noticed we should do this for `input-date-picker` as well (separately).

**Review by jcfranco — APPROVED** [review]
➡👁️ 🕸⬅

**Comment by jcfranco** [comment]
@eriklharper Can you give this a look as well and LMK if you spot anything? I can address any comments in a follow-up issue/PR.

**Comment by eriklharper** [comment]
> @eriklharper Can you give this a look as well and LMK if you spot anything? I can address any comments in a follow-up issue/PR.

This looks really good, nothing jumps out at me as weird.  Nice work! :)

**Pull request merged**